src/verilog/verilog_elaborate.cpp

```diff
@@ -365,6 +447,57 @@ void verilog_typecheckt::collect_symbols(const verilog_declt &decl)
   }
   else if(decl_class == ID_function || decl_class == ID_task)
   {
+    symbolt symbol;
+
+    symbol.mode = mode;
+    symbol.module = module_identifier;
+    symbol.value = decl;
+
+    typet return_type;
+
+    if(decl_class == ID_function)
+      return_type = convert_type(decl.type());
+    else
+      return_type = empty_typet();
+
+    symbol.type = code_typet{{}, return_type};
```

Comment:
Perhaps only create the `symbol` object right here with: `symbolt symbol{decl.get_identifier(), code_typet{{}, std::move(return_type)}, mode};`